lib/builders/ThemeBuilder.ts

```diff
@@ -115,23 +115,35 @@ interface MdThemingProvider {
 
 /** Angular Material Theme Builder interface */
 interface MdThemeBuilder {
-  primaryPalette: (value) => MdThemeBuilder;
-  accentPalette: (value) => MdThemeBuilder;
-  warnPalette: (value) => MdThemeBuilder;
-  backgroundPalette: (value) => MdThemeBuilder;
+  primaryPalette: (name, hue?: IThemeHues) => MdThemeBuilder;
+  accentPalette: (name, hue?: IThemeHues) => MdThemeBuilder;
+  warnPalette: (name, hue?: IThemeHues) => MdThemeBuilder;
+  backgroundPalette: (name, hue?: IThemeHues) => MdThemeBuilder;
   dark: () => MdThemeBuilder;
 }
 
 /** Angular Material Theme definition */
 export interface MdTheme {
   name?: string;
   primaryPalette?: string;
+  primaryPaletteHues?: IThemeHues;
   accentPalette?: string;
+  accentPaletteHues?: IThemeHues;
   warnPalette?: string;
+  warnPaletteHues?: IThemeHues;
   backgroundPalette?: string;
+  backgroundPaletteHues?: IThemeHues;
   dark?: boolean;
 }
 
+/** Angular Material Theme hue registry */
+export interface IThemeHues {
```

Comment:
Thank you very much! Yeah I don't think we should support those things in the Command Line, so we can leave that.